packages/@expo/cli/src/export/persistMetroAssets.ts

```diff
@@ -45,6 +45,11 @@ export async function persistMetroAssetsAsync(
     return;
   }
 
+  // For iOS, we need to ensure that the outputDirectory actually exists
+  if (platform === 'ios' && !fs.existsSync(outputDirectory)) {
+    fs.mkdirSync(outputDirectory, { recursive: true });
+  }
```

Comment:
```suggestion
  // For iOS, we need to ensure that the outputDirectory exists
  // because the bundle code and images build phase script does not check if it exists
  if (platform === 'ios' && !fs.existsSync(outputDirectory)) {
    fs.mkdirSync(outputDirectory, { recursive: true });
  }
```